src/Google.Api.Gax/Platform.cs

```diff
@@ -320,23 +316,26 @@ public static GkePlatformDetails TryLoad(string metadataJson, KubernetesData kub
             JObject metadata = null;
             JObject namespaceData = null;
             JObject podData = null;
-            try
+            // Parse JSON, ignoring all errors; partially available data is supported
+            try { metadata = JObject.Parse(metadataJson); } catch { }
+            try { namespaceData = JObject.Parse(kubernetesData.NamespaceJson); } catch { }
+            try { podData = JObject.Parse(kubernetesData.PodJson); } catch { }
+            if (metadata == null)
             {
-                metadata = JObject.Parse(metadataJson);
-                namespaceData = JObject.Parse(kubernetesData.NamespaceJson);
-                podData = JObject.Parse(kubernetesData.PodJson);
+                // Metadata is required. If it's not present, or the JSON cannot be parsed, return null.
+                return null;
             }
-            catch
+            if (namespaceData != null && namespaceData["kind"].Value<string>() != "Namespace")
```

Comment:
`?.Value`? Ditto for `podData` check below?

Reply:
Oops, yes done.